sdk/communication/azure-communication-callautomation/tests/test_hmac.py

```diff
@@ -0,0 +1,63 @@
+# coding: utf-8
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import unittest
+from azure.communication.callautomation._shared.policy import HMACCredentialsPolicy
+from devtools_testutils import AzureTestCase
+from azure.communication.callautomation._shared.utils import get_current_utc_time
+from azure.core.pipeline import PipelineRequest, PipelineContext
+from azure.core.rest import HttpRequest
+
+class HMACTest(AzureTestCase):
+    def setUp(self):
+        super(HMACTest, self).setUp()
+
+    def test_correct_hmac(self):
+        auth_policy = HMACCredentialsPolicy("https://contoso.communicationservices.azure.com", "pw==")
+
+        sha_val = auth_policy._compute_hmac("banana")
+        #[SuppressMessage("Microsoft.Security", "CS002:SecretInNextLine", Justification="unit test with fake keys")]
+        assert sha_val == "88EC05aAS9iXnaimtNO78JLjiPtfWryQB/5QYEzEsu8="
+
+    def test_correct_utf16_hmac(self):
+        auth_policy = HMACCredentialsPolicy("https://contoso.communicationservices.azure.com", "pw==")
+
+        sha_val = auth_policy._compute_hmac(u"😀")
+        #[SuppressMessage("Microsoft.Security", "CS002:SecretInNextLine", Justification="unit test with fake keys")]
+        assert sha_val == "1rudJKjn2Zi+3hRrBG29wIF6pD6YyAeQR1ZcFtXoKAU=" 
+
+    def mocked_get_current_utc_time():
+        return "Wed, 13 Apr 2022 18:09:12 GMT"
+
+    @unittest.mock.patch('azure.communication.callautomation._shared.policy.get_current_utc_time', mocked_get_current_utc_time)
+    def test_correct_signature(self):
+        auth_policy = HMACCredentialsPolicy("https://contoso.communicationservices.azure.com", "pw==")
+
+        test_request = HttpRequest(method = "GET", url = "https://contoso.communicationservices.azure.com/")
```

Comment:
39-41 lines are duplicated in a couple of test cases, maybe we can create a method with params http mehod& url?